packages/vite/src/node/server/searchRoot.ts

```diff
@@ -27,8 +27,12 @@ function hasWorkspacePackageJSON(root: string): boolean {
   if (!isFileReadable(path)) {
     return false
   }
-  const content = JSON.parse(fs.readFileSync(path, 'utf-8')) || {}
-  return !!content.workspaces
+  try {
+    const content = JSON.parse(fs.readFileSync(path, 'utf-8')) || {}
+    return !!content.workspaces
+  } catch {
+    return false
+  }
```

Comment:
`fs.readFileSync` might throw an error if `path` is not a file (`isFileReadable` doesn't check whether `path` is a file or not).
`JSON.parse` might throw an error if the content is not a valid JSON.

Reply:
Could we remove the `if (!isFileReadable(path)) {` check directly in this case since we can rely on the try catch instead?

Reply:
I think it's better to keep it. `isFileReadable` will filter the popular case (file doesn't exist) and we can avoid the slow try-catch.
It only affects if the project root is deep, but I think it's worth as this is only three lines of code.